### PR TITLE
Supports openharmony for v2.x

### PR DESCRIPTION
**Pull request by bu-wan**

### CODE REVIEW

File: cocos/audio/android/AudioEngine-inl.cpp [hunk 1]
```diff
@@ -104,7 +133,14 @@ static int fdGetter(const std::string& url, off_t* start, off_t* length)
         fd = AAsset_openFileDescriptor(asset, start, length);
         AAsset_close(asset);
     }
-
+    #elif CC_TARGET_PLATFORM == CC_PLATFORM_OPENHARMONY
+    FileUtilsOpenHarmony* fileUtils = dynamic_cast<FileUtilsOpenHarmony*>(FileUtils::getInstance());
```
Comment (minggo) [inline cocos/audio/android/AudioEngine-inl.cpp]: Can use static_cast?

File: cocos/audio/android/UrlAudioPlayer.cpp [hunk 2]
```diff
@@ -308,8 +310,8 @@ bool UrlAudioPlayer::prepare(const std::string &url, SLuint32 locatorType, std::
         return false;
     }
 
-    ALOGV("UrlAudioPlayer::prepare: %s, %s, %d, %d, %d", _url.c_str(), locatorTypeStr, _assetFd->getFd(), start,
-         length);
+     ALOGV("UrlAudioPlayer::prepare: %s, %s, %d, %d, %d", _url.c_str(), locatorTypeStr, _assetFd->getFd(), start,
+          length);
```
Comment (minggo) [inline cocos/audio/android/UrlAudioPlayer.cpp]: extra space

File: cocos/base/Macros.h [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+****************************************************************************/
+
+#pragma once
+
+#include <cstdint> // To include uint8_t, uint16_t and so on.
```
Comment (minggo) [inline cocos/base/Macros.h]: Is this file needed?

File: cocos/base/std/container/array.h [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+****************************************************************************/
+
+#pragma once
+
+#include <array>
+
+namespace ccstd {
+using std::array;
```
Comment (minggo) [inline cocos/base/std/container/array.h]: The same as above.

File: cocos/base/std/container/string.h [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+#pragma once
+
+#include <string>
+#include "boost/container/pmr/polymorphic_allocator.hpp"
+
+namespace ccstd {
+using std::string;
+
+namespace pmr {
+using string = std::basic_string<char, std::char_traits<char>, boost::container::pmr::polymorphic_allocator<char>>;
```
Comment (minggo) [inline cocos/base/std/container/string.h]: The same as above. Can just use stl directly in v2.4.12 and don't need to depend on boost.

File: cocos/base/std/optional.h [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    #include "base/std/container/string.h"
+    #include "boost/none.hpp"
+    #include "boost/optional.hpp"
+
+namespace ccstd {
+
+using boost::optional;
+using nullopt_t = boost::none_t;
+
+const nullopt_t nullopt((boost::none_t::init_tag())); // NOLINT // use std style
```
Comment (minggo) [inline cocos/base/std/optional.h]: The same as above.

File: cocos/platform/CCApplication.h [hunk 7]
```diff
@@ -55,6 +55,13 @@ class CC_DLL Application
         ANDROIDOS,   /**< Android, because ANDROID is a macro, so use ANDROIDOS instead */
         IPHONE,      /**< iPhone */
         IPAD,        /**< iPad */
+        BLACKBERRY,  /**< BLACKBERRY> */
+        NACL,        /**according to CCSys.js below*/
+        EMSCRIPTEN,
+        TIZEN,
+        WINRT,
+        WP8,
+        OpenHarmony, /**< OpenHarmony£¬because OPENHARMONY is a macro, so use OpenHarmony instead > */
```
Comment (minggo) [inline cocos/platform/CCApplication.h]: - only `OpenHarmony` is needed
- error characters: `OpenHarmony£¬because`

File: cocos/platform/openharmony/OpenHarmonyPlatform.cpp [hunk 8]
```diff
@@ -0,0 +1,335 @@
...
+//     se->start();
+    
+//     se::AutoHandleScope hs;
+//     // jsb_run_script("jsb-adapter/jsb-builtin.js");
+//     // jsb_run_script("main.js");
+    
+//     se->addAfterCleanupHook([]() {
+//         JSBClassType::destroy();
+//     });
+    // OpenHarmonyPlatform* platform = OpenHarmonyPlatform::getInstance();
```
Comment (minggo) [inline cocos/platform/openharmony/OpenHarmonyPlatform.cpp]: Can remove unneeded codes?

File: cocos/platform/openharmony/OpenHarmonyPlatform.cpp [hunk 9]
```diff
@@ -0,0 +1,335 @@
...
+// #include "application/ApplicationManager.h"
+// #include "application/CocosApplication.h"
+// #include "platform/UniversalPlatform.h"
+
+// #include "platform/openharmony/modules/System.h"
+// #include "platform/empty/modules/Screen.h"
+// #include "platform/empty/modules/Accelerometer.h"
+// #include "platform/empty/modules/Battery.h"
+// #include "platform/empty/modules/Network.h"
+// #include "platform/empty/modules/Vibrator.h"
```
Comment (minggo) [inline cocos/platform/openharmony/OpenHarmonyPlatform.cpp]: It is better to remove unneeded codes.

File: cocos/platform/openharmony/OpenHarmonyPlatform.cpp [hunk 10]
```diff
@@ -0,0 +1,335 @@
...
+        EventDispatcher::dispatchTickEvent(dt);
+        PoolManager::getInstance()->getCurrentPool()->clear();
+        now = std::chrono::steady_clock::now();
+        dt = std::chrono::duration_cast<std::chrono::microseconds>(now - prevTime).count() / 1000000.f;
+        prevTime = std::chrono::steady_clock::now();
+
+}
+
+
+
```
Comment (minggo) [inline cocos/platform/openharmony/OpenHarmonyPlatform.cpp]: extra spaces

File: cocos/platform/openharmony/OpenHarmonyPlatform.h [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+    uv_async_t _messageSignal{};
+    WorkerMessageQueue _messageQueue;
+    EGLCore* eglCore_{nullptr};
+
+    uint64_t width_;
+    uint64_t height_;
+    Application* g_app = nullptr;
+    //game started
+    bool g_started = false;
+
```
Comment (minggo) [inline cocos/platform/openharmony/OpenHarmonyPlatform.h]: extra space

File: CMakeLists.txt [hunk 12]
```diff
@@ -0,0 +1,923 @@
...
+##### cocos2d
+list(APPEND CC_COCOS_SOURCES
+    ${CMAKE_CURRENT_LIST_DIR}/cocos/cocos2d.cpp
+    ${CMAKE_CURRENT_LIST_DIR}/cocos/cocos2d.h
+)
+
+##### platform
+list(APPEND CC_COCOS_SOURCES
+${CMAKE_CURRENT_LIST_DIR}/cocos/platform/CCPlatformDefine.h
+${CMAKE_CURRENT_LIST_DIR}/cocos/platform/openharmony/CCPlatformDefine-openharmony.h
```
Comment (qiuguohua) [inline CMakeLists.txt]: It's best to insert indentations.

File: CMakeLists.txt [hunk 13]
```diff
@@ -0,0 +1,923 @@
...
+${CMAKE_CURRENT_LIST_DIR}/cocos/platform/openharmony/WorkerMessageQueue.h
+${CMAKE_CURRENT_LIST_DIR}/cocos/platform/openharmony/WorkerMessageQueue.cpp
+${CMAKE_CURRENT_LIST_DIR}/cocos/platform/openharmony/FileUtils-openharmony.cpp
+${CMAKE_CURRENT_LIST_DIR}/cocos/platform/openharmony/FileUtils-openharmony.h
+${CMAKE_CURRENT_LIST_DIR}/cocos/platform/openharmony/CCGL-openharmony.h
+${CMAKE_CURRENT_LIST_DIR}/cocos/platform/openharmony/CCDevice-openharmony.cpp
+${CMAKE_CURRENT_LIST_DIR}/cocos/platform/openharmony/OpenHarmonyPlatform.cpp
+${CMAKE_CURRENT_LIST_DIR}/cocos/platform/openharmony/OpenHarmonyPlatform.h
+${CMAKE_CURRENT_LIST_DIR}/cocos/platform/openharmony/render/egl_core.cpp
+${CMAKE_CURRENT_LIST_DIR}/cocos/platform/openharmony/render/egl_core.h
```
Comment (qiuguohua) [inline CMakeLists.txt:54]: We didn't use the egl_core file?

File: predefine.cmake [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+add_definitions(-DOPENHARMONY=${CC_PLATFORM_OPENHARMONY})
+add_definitions(-DCC_PLATFORM=${CC_PLATFORM})
+set(CMAKE_CXX_FLAGS "-fvisibility=hidden -fvisibility-inlines-hidden ${CMAKE_CXX_FLAGS}")
+if("${OHOS_ARCH}" STREQUAL "armeabi-v7a")
+    set(CMAKE_CXX_FLAGS "-march=armv7a ${CMAKE_CXX_FLAGS}")
+endif()
+
+MESSAGE(STATUS "platform: ${CMAKE_SYSTEM_NAME}")
+
+################################# cc_set_if_undefined ###################################
```
Comment (minggo) [inline predefine.cmake]: I think it is not needed as v2.x doesn't use cmake.

File: cocos/audio/AudioEngine.cpp [hunk 15]
```diff
@@ -365,7 +365,7 @@ void AudioEngine::onPause(const CustomEvent &event) {
         }
     }
     
-#if CC_TARGET_PLATFORM == CC_PLATFORM_ANDROID
+#if CC_TARGET_PLATFORM == CC_PLATFORM_ANDROID ||  CC_TARGET_PLATFORM == CC_PLATFORM_OPENHARMONY
```
Comment (qiuguohua) [inline cocos/audio/AudioEngine.cpp]: `#if CC_TARGET_PLATFORM == CC_PLATFORM_ANDROID || CC_TARGET_PLATFORM == C_PLATFORM_OPENHARMONY`

File: templates/openharmony/native/engine/build.bat [hunk 16]
```diff
@@ -0,0 +1,81 @@
+@echo off 
+:begin
+echo start build
```
Comment (minggo) [inline templates/openharmony/native/engine/build.bat]: Is it needed after integrating with editor?

File: templates/openharmony/native/engine/build.sh [hunk 17]
```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+buildMode="bc"
+
+enginePath="E:\CocosDashboard_1.2.5\resources\.editors\Creator\2.4.11\CocosCreator.exe"
```
Comment (minggo) [inline templates/openharmony/native/engine/build.sh]: The same as above.

File: templates/openharmony/native/engine/gulpfile.js [hunk 18]
```diff
@@ -0,0 +1,277 @@
+const gulp = require('gulp');
+const Del = require('del');
+const through = require('through2');
+const minimist = require('minimist');
```
Comment (minggo) [inline templates/openharmony/native/engine/gulpfile.js]: Is this file needed?

File: templates/openharmony/native/engine/gulpfile.js [hunk 19]
```diff
@@ -0,0 +1,277 @@
...
+        case 'reversePortrait':// 竖屏倒转
+            jsonOrientation = "portrait_inverted";
+            break;
+        case 'reverseLandscape':// 横屏倒转
+            jsonOrientation = "landscape_inverted";
+            break;
+        case 'sensorLandscape'://  横向显示，但是基于设备传感器
+            jsonOrientation = "auto_rotation_landscape";
+            break;
+        case 'sensorPortrait'://  竖向显示，但是基于设备传感器
```
Comment (minggo) [inline templates/openharmony/native/engine/gulpfile.js]: Please don't use Chinese.

File: templates/openharmony/native/engine/手动构建说明.md [hunk 20]
```diff
@@ -0,0 +1,112 @@
+# 手动操作指南
+## 一.环境配置
+下载DevEco开发工具并配置相关环境:
```
Comment (minggo) [inline templates/openharmony/native/engine/手动构建说明.md]: I think it is not needed. Cocos will release docs after integrating with editor.

File: templates/openharmony/native/engine/自动化构建说明.md [hunk 21]
```diff
@@ -0,0 +1,92 @@
+# 自动化构建工具操作指南
+
+### 一.环境配置
+
+#### 1. 下载DevEco开发工具并配置相关环境
+* 参考文档：
+[DevEco指南](https://developer.harmonyos.com/cn/docs/documentation/doc-guides-V3/installation_process-0000001071425528-V3)
+* 对于大型游戏需要配置环境变量设置DevEco编译内存
+NODE_OPTIONS
+--max_old_space_size=10240
```
Comment (minggo) [inline templates/openharmony/native/engine/自动化构建说明.md]: The same as above.

**Comment by minggo** [comment]
I think the files are not needed:
- files in sign_tools
- files in ohosTest
- files in hvigor
- files in engine/code

File: cocos/audio/android/utils/Utils.cpp [hunk 22]
```diff
@@ -33,7 +36,12 @@ namespace cocos2d {
 
 int getSDKVersion()
 {
-    return JniHelper::callStaticIntMethod(JCLS_HELPER, "getSDKVersion");
+#if CC_TARGET_PLATFORM == CC_PLATFORM_ANDROID
+     return JniHelper::callStaticIntMethod(JCLS_HELPER, "getSDKVersion");
+#elif CC_TARGET_PLATFORM == CC_PLATFORM_OPENHARMONY
+    // openharmony未实现
```
Comment (qiuguohua) [inline cocos/audio/android/utils/Utils.cpp]: Needs to be aligned. We can use todo.

File: cocos/network/HttpClient.cpp [hunk 23]
```diff
@@ -174,11 +174,11 @@ static bool configureCURL(HttpClient* client, HttpRequest* request, CURL* handle
     if (code != CURLE_OK) {
         return false;
     }
-    code = curl_easy_setopt(handle, CURLOPT_TIMEOUT, request->getTimeout());
+    code = curl_easy_setopt(handle, CURLOPT_TIMEOUT, (long)request->getTimeout());
```
Comment (qiuguohua) [inline cocos/network/HttpClient.cpp]: Recommended to use static_cast

File: cocos/platform/openharmony/CCDevice-openharmony.cpp [hunk 24]
```diff
@@ -0,0 +1,151 @@
...
+
+void Device::vibrate(float duration)
+{
+    int32_t value = 0;
+    NapiHelper::napiCallFunctionByFloatArgs("vibrate", duration, &value);
+}
+
+NS_CC_END
+
+#endif // CC_TARGET_PLATFORM == CC_PLATFORM_ANDROID
```
Comment (qiuguohua) [inline cocos/platform/openharmony/CCDevice-openharmony.cpp]: It's the Openharmony platform.

File: cocos/platform/openharmony/CCDevice-openharmony.cpp [hunk 25]
```diff
@@ -0,0 +1,151 @@
...
+int Device::getDevicePixelRatio() {
+    // float value;
+    // NapiHelper::napiCallFunction("getPixelRation", &value);
+    // return value;
+    // TODO(qgh):openharmony does support this interface, but returning a value of 1.5 will cause the entire page to zoom in.
+    return 1;
+}
+
+void Device::setKeepScreenOn(bool value)
+{
```
Comment (qiuguohua) [inline cocos/platform/openharmony/CCDevice-openharmony.cpp]: `{` should not be on a separate line

File: cocos/platform/openharmony/OpenHarmonyPlatform.cpp [hunk 26]
```diff
@@ -0,0 +1,335 @@
...
+        uv_timer_start(&_timerHandle, &OpenHarmonyPlatform::timerCb, 0, 1);
+     }
+}
+
+int32_t OpenHarmonyPlatform::loop() {
+    return 0;
+}
+
+void OpenHarmonyPlatform::onSurfaceCreated(OH_NativeXComponent* component, void* window) {
+    eglCore_ = new EGLCore();
```
Comment (qiuguohua) [inline cocos/platform/openharmony/OpenHarmonyPlatform.cpp:251]: Why do we need to use EGLCore?

File: cocos/scripting/js-bindings/jswrapper/Value.cpp [hunk 27]
```diff
@@ -484,6 +511,23 @@ namespace se {
...
+        	// CC_LOG_WARNING("convert int64 to double");
+        	return static_cast<double>(_u._bigint);
+    	}
+
+    	if (_type == Type::String) {
+        	return std::stod(*_u._string);
+    	}
+
+    	return _u._boolean ? 1.0 : 0.0;
+	}
```
Comment (qiuguohua) [inline cocos/scripting/js-bindings/jswrapper/Value.cpp]: Incorrect code format

File: cocos/scripting/js-bindings/jswrapper/Value.hpp [hunk 28]
```diff
@@ -47,7 +47,8 @@ namespace se {
             Number,
             Boolean,
             String,
-            Object
+            Object,        // NOLINT(readability-identifier-naming)
+        	BigInt
```
Comment (qiuguohua) [inline cocos/scripting/js-bindings/jswrapper/Value.hpp]: extra space

File: cocos/scripting/js-bindings/jswrapper/Value.hpp [hunk 29]
```diff
@@ -237,6 +238,24 @@ namespace se {
          *  @param[in] v The double value to be set.
          */
         void setNumber(double v);
+		
+		/**
```
Comment (qiuguohua) [inline cocos/scripting/js-bindings/jswrapper/Value.hpp]: extra space

File: cocos/scripting/js-bindings/jswrapper/Value.hpp [hunk 30]
```diff
@@ -324,6 +345,12 @@ namespace se {
          */
         double toNumber() const;
 
+		/**
```
Comment (qiuguohua) [inline cocos/scripting/js-bindings/jswrapper/Value.hpp]: extra space

File: cocos/scripting/js-bindings/jswrapper/Value.hpp [hunk 31]
```diff
@@ -361,6 +388,12 @@ namespace se {
          *  @return true if se::Value stores a number, otherwise false.
          */
         inline bool isNumber() const { return _type == Type::Number; }
+		
+		 /**
```
Comment (qiuguohua) [inline cocos/scripting/js-bindings/jswrapper/Value.hpp]: extra space

File: cocos/scripting/js-bindings/jswrapper/config.hpp [hunk 32]
```diff
@@ -49,14 +50,14 @@
             #define SCRIPT_ENGINE_TYPE           SCRIPT_ENGINE_JSC
         #endif
     #endif
-
+#elif defined(OPENHARMONY)
```
Comment (qiuguohua) [inline cocos/scripting/js-bindings/jswrapper/config.hpp:53]: Why not use `CC_TARGET_PLATFORM == CC_PLATFORM_OPENHARMONY` ?

File: CMakeLists.txt [hunk 33]
```diff
@@ -0,0 +1,923 @@
...
+################################# cocos engine source code ################################
+
+##### cocos2d
+list(APPEND CC_COCOS_SOURCES
+    ${CMAKE_CURRENT_LIST_DIR}/cocos/cocos2d.cpp
+    ${CMAKE_CURRENT_LIST_DIR}/cocos/cocos2d.h
+)
+
+##### platform
+list(APPEND CC_COCOS_SOURCES
```
Comment (qiuguohua) [inline CMakeLists.txt:53]: Why not use `cocos_source_files`?

File: cocos/audio/android/AudioDecoderSLES.cpp [hunk 34]
```diff
@@ -463,6 +466,7 @@ bool AudioDecoderSLES::decodeToPcm()
 
     std::string info = _result.toString();
     ALOGI("Original audio info: %s, total size: %d", info.c_str(), (int)_result.pcmBuffer->size());
+   #endif
```
Comment (qiuguohua) [inline cocos/audio/android/AudioDecoderSLES.cpp]: extra space

File: cocos/audio/android/AudioEngine-inl.cpp (hunk 1 above)
Comment (bu-wan) [reply cocos/audio/android/AudioEngine-inl.cpp]: fixed

File: cocos/audio/android/UrlAudioPlayer.cpp (hunk 2 above)
Comment (bu-wan) [reply cocos/audio/android/UrlAudioPlayer.cpp]: fixed

File: cocos/base/std/optional.h (hunk 6 above)
Comment (bu-wan) [reply cocos/base/std/optional.h]: fixed

File: cocos/platform/openharmony/OpenHarmonyPlatform.cpp (hunk 8 above)
Comment (bu-wan) [reply cocos/platform/openharmony/OpenHarmonyPlatform.cpp]: fixed

File: cocos/platform/openharmony/OpenHarmonyPlatform.cpp (hunk 9 above)
Comment (bu-wan) [reply cocos/platform/openharmony/OpenHarmonyPlatform.cpp]: fixed

File: cocos/platform/openharmony/OpenHarmonyPlatform.cpp (hunk 10 above)
Comment (bu-wan) [reply cocos/platform/openharmony/OpenHarmonyPlatform.cpp]: fixed

File: cocos/base/std/container/string.h (hunk 5 above)
Comment (bu-wan) [reply cocos/base/std/container/string.h]: use stl string

File: predefine.cmake (hunk 14 above)
Comment (bu-wan) [reply predefine.cmake]: merge  it into CMakeLists.txt

File: templates/openharmony/native/engine/build.bat (hunk 16 above)
Comment (bu-wan) [reply templates/openharmony/native/engine/build.bat]: it is deleted

File: templates/openharmony/native/engine/build.sh (hunk 17 above)
Comment (bu-wan) [reply templates/openharmony/native/engine/build.sh]: it is deleted

File: templates/openharmony/native/engine/gulpfile.js (hunk 18 above)
Comment (bu-wan) [reply templates/openharmony/native/engine/gulpfile.js]: it is deleted

File: templates/openharmony/native/engine/gulpfile.js (hunk 19 above)
Comment (bu-wan) [reply templates/openharmony/native/engine/gulpfile.js]: it is deleted

File: templates/openharmony/native/engine/手动构建说明.md (hunk 20 above)
Comment (bu-wan) [reply templates/openharmony/native/engine/手动构建说明.md]: it is deleted

File: templates/openharmony/native/engine/自动化构建说明.md (hunk 21 above)
Comment (bu-wan) [reply templates/openharmony/native/engine/自动化构建说明.md]: it is deleted

**Comment by bu-wan** [comment]
>

**Comment by bu-wan** [comment]
> >

File: CMakeLists.txt (hunk 12 above)
Comment (bu-wan) [reply CMakeLists.txt]: fixed

File: cocos/audio/AudioEngine.cpp (hunk 15 above)
Comment (bu-wan) [reply cocos/audio/AudioEngine.cpp]: fixed

File: cocos/audio/android/utils/Utils.cpp (hunk 22 above)
Comment (bu-wan) [reply cocos/audio/android/utils/Utils.cpp]: fixed

File: cocos/network/HttpClient.cpp (hunk 23 above)
Comment (bu-wan) [reply cocos/network/HttpClient.cpp]: fixed

File: cocos/platform/openharmony/CCDevice-openharmony.cpp (hunk 24 above)
Comment (bu-wan) [reply cocos/platform/openharmony/CCDevice-openharmony.cpp]: fixed

File: cocos/platform/openharmony/CCDevice-openharmony.cpp (hunk 25 above)
Comment (bu-wan) [reply cocos/platform/openharmony/CCDevice-openharmony.cpp]: fixed

File: cocos/platform/openharmony/OpenHarmonyPlatform.cpp (hunk 26 above)
Comment (bu-wan) [reply cocos/platform/openharmony/OpenHarmonyPlatform.cpp:251]: EGLCore is new class that handle egl logic,not java eglcore

File: cocos/scripting/js-bindings/jswrapper/Value.cpp (hunk 27 above)
Comment (bu-wan) [reply cocos/scripting/js-bindings/jswrapper/Value.cpp]: fixed

File: cocos/scripting/js-bindings/jswrapper/Value.hpp (hunk 28 above)
Comment (bu-wan) [reply cocos/scripting/js-bindings/jswrapper/Value.hpp]: fixed

File: cocos/scripting/js-bindings/jswrapper/Value.hpp (hunk 29 above)
Comment (bu-wan) [reply cocos/scripting/js-bindings/jswrapper/Value.hpp]: fixed

File: cocos/scripting/js-bindings/jswrapper/Value.hpp (hunk 30 above)
Comment (bu-wan) [reply cocos/scripting/js-bindings/jswrapper/Value.hpp]: fixed

File: cocos/scripting/js-bindings/jswrapper/Value.hpp (hunk 31 above)
Comment (bu-wan) [reply cocos/scripting/js-bindings/jswrapper/Value.hpp]: fixed

File: CMakeLists.txt (hunk 33 above)
Comment (bu-wan) [reply CMakeLists.txt:53]: cocos_source_files is a func that is used in creator 3.x ,it is too complex for creator 2.x because cmake is only used for openharmony.

File: cocos/scripting/js-bindings/jswrapper/config.hpp (hunk 32 above)
Comment (liusiyuan777) [reply cocos/scripting/js-bindings/jswrapper/config.hpp:53]: Because the use of CC_PLATFORM_OPENHARMONY needs to include other .h files, which causes redundant include.

File: CMakeLists.txt [hunk 35]
```diff
@@ -0,0 +1,918 @@
...
+        ${CMAKE_CURRENT_LIST_DIR}/cocos/renderer/scene/assembler/SlicedSprite3D.cpp
+        ${CMAKE_CURRENT_LIST_DIR}/cocos/renderer/scene/assembler/SlicedSprite3D.hpp
+        ${CMAKE_CURRENT_LIST_DIR}/cocos/renderer/scene/assembler/TiledMapAssembler.cpp
+        ${CMAKE_CURRENT_LIST_DIR}/cocos/renderer/scene/assembler/TiledMapAssembler.hpp
+        ${CMAKE_CURRENT_LIST_DIR}/cocos/renderer/scene/MemPool.cpp
+        ${CMAKE_CURRENT_LIST_DIR}/cocos/renderer/scene/MemPool.hpp
+        ${CMAKE_CURRENT_LIST_DIR}/cocos/renderer/scene/MeshBuffer.cpp
+        ${CMAKE_CURRENT_LIST_DIR}/cocos/renderer/scene/MeshBuffer.hpp
+        ${CMAKE_CURRENT_LIST_DIR}/cocos/renderer/scene/ModelBatcher.cpp
+        ${CMAKE_CURRENT_LIST_DIR}/cocos/renderer/scene/ModelBatcher.hpp
```
Comment (minggo) [inline CMakeLists.txt:454]: 2.x doesn't have `cocos/renderer`. Please double check the files in this file.

File: cocos/base/std/container/string.h (hunk 5 above)
Comment (minggo) [reply cocos/base/std/container/string.h]: Can just use stl string in codes. Don't need `cocos/base/std` folder.

File: cocos/platform/openharmony/CCGL-openharmony.h [hunk 36]
```diff
@@ -0,0 +1,113 @@
...
+// // normal process
+// #include <GLES3/gl3.h>
+// #include <EGL/egl.h>
+// #include <EGL/eglext.h>
+// // gl2.h doesn't define GLchar on Android
+// typedef char GLchar;
+// // android defines GL_BGRA_EXT but not GL_BRGA
+// #ifndef GL_BGRA
+// #define GL_BGRA  0x80E1
+// #endif
```
Comment (minggo) [inline cocos/platform/openharmony/CCGL-openharmony.h]: Please  remove unneeded codes.

File: cocos/platform/openharmony/render/egl_core.h [hunk 37]
```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2021 Huawei Device Co., Ltd.
```
Comment (minggo) [inline cocos/platform/openharmony/render/egl_core.h]: 2021 -> 2023?

File: cocos/platform/openharmony/render/egl_core.cpp [hunk 38]
```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2021 Huawei Device Co., Ltd.
```
Comment (minggo) [inline cocos/platform/openharmony/render/egl_core.cpp]: 2021 -> 2023

File: cocos/base/Macros.h (hunk 3 above)
Comment (bu-wan) [reply cocos/base/Macros.h]: remove it

File: cocos/base/std/container/array.h (hunk 4 above)
Comment (bu-wan) [reply cocos/base/std/container/array.h]: remove it

File: cocos/base/std/container/string.h (hunk 5 above)
Comment (bu-wan) [reply cocos/base/std/container/string.h]: fixed

File: CMakeLists.txt (hunk 35 above)
Comment (bu-wan) [reply CMakeLists.txt:454]: 2.x has renderer ,which is different 3.x

File: cocos/ui/edit-box/EditBox-openharmony.cpp [hunk 39]
```diff
@@ -0,0 +1,115 @@
...
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+****************************************************************************/
+
+#include "EditBox.h"
+#include "EditBox-openharmony.h"
```
Comment (qiuguohua) [inline cocos/ui/edit-box/EditBox-openharmony.cpp:27]: The videoplay and editbox are now implemented on ts

File: cocos/platform/openharmony/CCGL-openharmony.h (hunk 36 above)
Comment (bu-wan) [reply cocos/platform/openharmony/CCGL-openharmony.h]: fixed

File: cocos/platform/openharmony/render/egl_core.h (hunk 37 above)
Comment (bu-wan) [reply cocos/platform/openharmony/render/egl_core.h]: fixed

File: cocos/platform/openharmony/render/egl_core.cpp (hunk 38 above)
Comment (bu-wan) [reply cocos/platform/openharmony/render/egl_core.cpp]: fixed

File: cocos/audio/android/AudioDecoderSLES.cpp (hunk 34 above)
Comment (bu-wan) [reply cocos/audio/android/AudioDecoderSLES.cpp]: delete extra space

**Review by minggo — APPROVED** [review]

File: templates/openharmony/native/engine/openharmony/entry/src/main/ets/workers/cocos_worker.ts [hunk 40]
```diff
@@ -0,0 +1,111 @@
+import hilog from '@ohos.hilog';
```
Comment (qiuguohua) [inline templates/openharmony/native/engine/openharmony/entry/src/main/ets/workers/cocos_worker.ts:1]: Can it be placed under copyright information?

**Comment by PPpro** [comment]
can we remove the whole 'ohosTest' directory? 

It seems we don't need this module in our DevEco project ?

**Comment by bu-wan** [comment]
> can we remove the whole 'ohosTest' directory?
> 
> It seems we don't need this module in our DevEco project ?



> 
fixed

**Review by PPpro — APPROVED** [review]

**Review by qiuguohua — APPROVED** [review]

File: cocos/scripting/js-bindings/jswrapper/config.hpp (hunk 32 above)
Comment (bu-wan) [reply cocos/scripting/js-bindings/jswrapper/config.hpp:53]: Because the use of CC_PLATFORM_OPENHARMONY needs to include other .h files, which causes redundant include.
